src/Cake.Frosting/AsyncFrostingTask.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Cake.Core;
+using Cake.Frosting.Internal;
+
+namespace Cake.Frosting
+{
+    /// <summary>
+    /// Base class for an asynchronous Frosting task using the standard context.
+    /// </summary>
+    /// <seealso cref="ICakeContext" />
+    public abstract class AsyncFrostingTask : AsyncFrostingTask<ICakeContext>
+    {
+    }
+
+    /// <summary>
+    /// Base class for an asynchronous Frosting task using a custom context.
+    /// </summary>
+    /// <typeparam name="T">The context type.</typeparam>
+    /// <seealso cref="IFrostingTask" />
+    public abstract class AsyncFrostingTask<T> : IFrostingTask
```

Comment:
If you don't like duplication between this and `FrostingTask`, we could have a look at introducing a common abstract class. My gut feel was the amount of code duplicated was not warranting, but happy to get your feel on this.